server/pxf-hive/src/main/java/org/greenplum/pxf/plugins/hive/HiveORCSearchArgumentBuilder.java

```diff
@@ -0,0 +1,328 @@
+        this.filterBuilder = SearchArgumentFactory.newBuilder(configuration);
+        this.columnDescriptors = tupleDescription;
+        this.hasLogicalOperators = false;
+    }
+
+    @Override
+    public Node before(Node node) {
+        if (node instanceof OperatorNode) {
+            OperatorNode operatorNode = (OperatorNode) node;
+            Operator operator = operatorNode.getOperator();
+            if (operator.isLogical()) {
+                // AND / OR / NOT
+                switch (operator) {
+                    case OR:
+                        filterBuilder.startOr();
+                        break;
+                    case AND:
+                        filterBuilder.startAnd();
+                        break;
+                    case NOT:
+                        filterBuilder.startNot();
+                        break;
+                }
+                hasLogicalOperators = true;
+            }
+        }
+        return node;
+    }
+
+    @Override
+    public Node visit(Node node) {
+        if (node instanceof OperatorNode) {
+            OperatorNode operatorNode = (OperatorNode) node;
+            Operator operator = operatorNode.getOperator();
+
+            if (!operator.isLogical()) {
+                if (!hasLogicalOperators) {
+                    /*
+                     * If there is only a single filter it will need special
+                     * case logic to make sure to still wrap the filter in a
+                     * startAnd() & end() block
+                     */
+                    filterBuilder.startAnd();
+                }
+                buildArgument(operatorNode);
+                if (!hasLogicalOperators) {
+                    filterBuilder.end();
+                }
+            }
+        }
+        return node;
+    }
+
+    @Override
+    public Node after(Node node) {
+        if (node instanceof OperatorNode) {
+            OperatorNode operatorNode = (OperatorNode) node;
+            if (operatorNode.getOperator().isLogical()) {
+                // AND / OR / NOT
+                filterBuilder.end();
+            }
+        }
+        return node;
+    }
+
+    public SearchArgument.Builder getFilterBuilder() {
+        return filterBuilder;
+    }
+
+    /**
+     * Builds a single argument
+     *
+     * @param operatorNode the operator node
+     * @return true if the argument is build, false otherwise
+     */
+    private boolean buildArgument(OperatorNode operatorNode) {
+
+        Operator operator = operatorNode.getOperator();
+        ColumnIndexOperand columnIndexOperand = operatorNode.getColumnIndexOperand();
+        Optional<Operand> valueOperand = operatorNode.getOperand();
+
+        ColumnDescriptor filterColumn = columnDescriptors.get(columnIndexOperand.index());
+        String filterColumnName = filterColumn.columnName();
+        Object filterValue = null;
+
+        if (valueOperand.isPresent()) {
```

Comment:
```suggestion
        if (valueOperand != null) {
```
if Optional gets pushed into operand